src/sem.c

```diff
@@ -2245,11 +2245,33 @@ static void sem_check_static_elab(tree_t t)
    }
 }
 
+static bool sem_check_missing_wait(tree_t t, nametab_t *tab) {
+   const int ntriggers = tree_triggers(t);
+   if (ntriggers > 0)
+      return true;
+   
+   const tree_flags_t flags = tree_flags(t);
+   if (flags & TREE_F_HAS_WAIT)
+      return true;
+   
+   diag_t *d = diag_new(DIAG_WARN, tree_loc(t));
+   diag_printf(d, "potential infinite loop in process");
+   if (!(flags & TREE_F_SYNTHETIC_NAME))
+      diag_printf(d, " %s", istr(tree_ident(t)));
+   diag_printf(d, " with no sensitivity list and no wait statements.");
+   diag_emit(d);
+
+   return true;
+}
+
 static bool sem_check_process(tree_t t, nametab_t *tab)
 {
    if (!sem_check_sensitivity(t, tab))
       return false;
 
+   if (opt_get_int(OPT_MISSING_WAIT) && !sem_check_missing_wait(t, tab))
```

Comment:
I'd put the `OPT_MISSING_WAIT` check inside `sem_check_missing_wait`. Seems cleaner that way.